torch/_dynamo/variables/torch.py

```diff
@@ -476,6 +476,22 @@ def get_state_from_generator():
                 **options,
             )
 
+            def track_out_tensors(new_tensor_variable, old_tensor_variable):
+                local_name = tx.find_symbolic_locals_name(old_tensor_variable)
+                if local_name:
+                    tx.symbolic_locals[local_name] = new_tensor_variable
+                    return
+
+                global_name = tx.find_global_name(old_tensor_variable)
+                if global_name and global_name in tx.f_globals:
+                    # We are writing into a global variable, so track mutation.
+                    tx.track_side_effect_for_globals(global_name, new_tensor_variable)
```

Comment:
This is not a pre-existing condition and it looks pretty wrong. The code you are reusing here is what you'd use for `STORE_GLOBAL`, but an `f(out=x)` is different than `global x; x = ...` because of treatment of aliases. Consider `y = x` beforehand; you need y to reflect the metadata change (e.g., if x was zero size and got resized as part of out) on all aliases including y.